Allow user to specify custom config directory and specify the active user when calling .bat or .sh file

SomeOddCodeGuy opened the pull request:
This is to allow a user to call the .bat file, .sh file or .py file with arguments to specify the config directory or the user.

With this change, you can either call the .bat file or .sh file like normal and it will use the default config directory within Wilmer itself, or you can pass the arguments in.

### Windows

**Use config folder in Wilmer directory like normal**

`run_windows.bat`

**Use custom config location and specify user using positional arguments**

`run_windows.bat "C:\Temp" "TestUser"`

**Use custom config location and specify user using named arguments**

`run_windows.bat --ConfigDirectory "C:\Temp" --User "TestUser"`

**Send only the user**

`run_windows.bat --User "TestUser"`

###MacOS

**Use config folder in Wilmer directory like normal**

`./run_mac.sh`

**Use custom config location and specify user using positional arguments**

`./run_mac.sh "/path/to/config" "TestUser"`

**Use custom config location and specify user using named arguments**

`./run_mac.sh --ConfigDirectory="/path/to/config" --User="TestUser"`

**Send only the user**

`./run_mac.sh --User="TestUser"`

###Python Script

**Use config folder in Wilmer directory like normal (Windows example)**

`python server.py`

**Use custom config location and specify user using positional arguments**

`python server.py "/path/to/config" "TestUser"`

**Use custom config location and specify user using named arguments**

`python server.py --ConfigDirectory="/path/to/config" --User="TestUser"`

**Send only the user**

`python server.py --User="TestUser"`

EDIT:

Updated the SqlLite db to be created for each user. Additionally, you can now specify the directory that the sqllite db will be created.

Here is an example from a user config .json file:

```json
...
  "discussionDirectory": "D:\\temp",
  "sqlLiteDirectory": "D:\\temp\\WilmerSqlLite",
...
```

In the event that property does not exist, it'll use the root directory of Wilmer.

jeffqg commented on run_macos.sh:
```diff
@@ -27,9 +36,8 @@ if [ $? -ne 0 ]; then
-echo "Starting the application..."
-python server.py
+echo "Starting the application with ConfigDirectory=$CONFIG_DIR and User=$USER..."
+python3 server.py "$CONFIG_DIR" "$USER"
```
On Linux, you should be able to leave all the argument parsing out of the runner script and just call `python3 server.py "$@"`, letting server.py do all the parsing. 

I've been using run_macos.sh on Fedora with zsh and it's work flawlessly. Hopefully the above suggestions works on MacOS, but I have no clue how it would be done on Windows.

SomeOddCodeGuy replied:
I was thinking of separating out a mac and linux sh for stuff like that in the past, but now I'm torn. On one hand, that would be cleaner for Linux to use your syntax (assuming it doesn't work for Mac... it might), but on the other hand the fact that you're able to use run_macos.sh makes me want to just rename that file "run_wilmer.sh" and have windows be "run_wilmer.bat".

Either way, let me know how the SH does in its current state. I've had more chance to test this on Windows than on Mac so I could have broken something in .sh file.

SomeOddCodeGuy replied:
I tried your suggestion on Mac and it worked. I've updated the bash script with it. If it works for you, I'm just gonna rename that file "run_wilmer.sh" for both mac and linux

jeffqg commented on server.py:
You might want to use [argparse](https://docs.python.org/3/library/argparse.html) to handle all this logic for you.

SomeOddCodeGuy replied:
Swapped to argparse!

SomeOddCodeGuy commented:
I added something to the PR just now to handle the sqllite db. Look back at the bottom of the original comment for more info, but you can now specify a custom directory for your sqllite db as well. 

Also pull down latest master if you had grabbed it after my PRs were done. I had broken something on accident and just fixed it. The latest master is pulled into this branch so if you're using this then you should have everything.

jeffqg commented:
Everything is looking good on my side. I'm using the "$@" version of the run_macos.sh script, with a config directory & user passed in.
